src/Microsoft.Extensions.ApiDescription.Client/build/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -235,6 +245,13 @@
   <!-- ServiceFileReference support -->
 
   <Target Name="_ServiceFileReferenceGenerator_GetMetadata" Condition="'@(ServiceFileReference)' != ''">
+    <!-- Repeat default namespace checks in case $(RootNamespace) is defined after .props file is imported. -->
+    <PropertyGroup>
+      <ServiceFileReferenceCSharpNamespace
+          Condition="'$(ServiceFileReferenceCSharpNamespace)' == ''">$(RootNamespace)</ServiceFileReferenceCSharpNamespace>
+      <ServiceFileReferenceTypeScriptNamespace
+          Condition="'$(ServiceFileReferenceTypeScriptNamespace)' == ''">$(RootNamespace)</ServiceFileReferenceTypeScriptNamespace>
+    </PropertyGroup>
```

Comment:
So when I see things like this, I get a little nervous. The F# folks like add send PRs to add support for their own stuff. Is there a straightforward way to do this without hardcoding language names into MSBuild properties?

Is this just here so you can configure a different namespace for C# and typescript?

Reply:
> Is this just here so you can configure a different namespace for C# and typescript?

Exactly

Reply:
You can still configure a namespace independently for each generator though right? Is it going to be common for a project to do both? It seems way less surprising to me to have a single 'default' namespace setting, and then you can configure it per-reference if you need to.

Reply:
Yes, you can still configure a namespace per `@(ServiceProjectReference)`, `@(ServiceUriReference)` or `@(ServiceFileReference)` item. If someone wanted to change the namespace per-generator, the generator would need to use a unique property (that defaults to `$(RootNamespace)`) to support them.

But, are you saying per-generator namespaces are not a requirement? If that's the case, I could remove the per-language namespace properties and change the default namespace (set in `GetFileReferenceMetadata`) to `$(RootNamespace)`.